### PR TITLE
Gorger buff of the year (RUTGMC port)

### PR DESCRIPTION
**Pull request by Blundir**
## About The Pull Request

- Feast now works on all xenos in 4 tile radius, cooldown decreased from 180 to 30
- Feast healing now has visual effect when it's active
- Drain and Carnage drain real blood from people (30u)
- Transfusion range increased from 2 tiles to 3
- You can move while casting Transfusion
- Psychic Link cooldown from 50 to 15
- Psychic Link doesn't require to stand still
- Gorger now has passive plasma regen up to 10% of maximum plasma

dreamseeker_41t1jVpGkr

100% port of https://github.com/Cosmic-Overlord/TerraGov-Marine-Corps/pull/122 and https://github.com/Cosmic-Overlord/TerraGov-Marine-Corps/pull/129

## Why It's Good For The Game

Gorger is really uncomfortable to play in general, these are merely QoL considering what state he is in right now, Gorger cannot sustain effectively neither offensive nor support abilities, some abilities are very rarely used in general, this tries to correct that (very successfully, considering RU experience with it). Probably needs a testmerge to estimate how good/bad these changes are. These changes are present on RU for 3 months without any balance issues.

## Changelog
:cl:
add: Drain and Carnage drain real blood from people (30u)
add: Feast now works on all xenos in 4 tile radius
qol: gorger can cast transfusion while moving
qol: gorger regenerates 10% of blood automatically
qol: Psychic Link doesn't require to stand still
balance: Transfusion range increased from 2 tiles to 3
balance: Feast cooldown lowered  from 180 to 30
balance: Psychic Link cooldown lowered from 50 to 15
/:cl:

### CODE REVIEW

**Comment by lKiseki** [comment]
Woah woah woah that's one too many eggs in one basket. Gorger doesn't even _need_ half of these.

**Comment by Blundir** [comment]
> > Drain and Carnage drain real blood from people (30u)
> 
> Unlikely will pass since QV has plans for blood.
> 
QV seen my plans fod gorger and haven't said anything about blood

**Comment by blackdav123** [comment]
feast actually working as a support ability with the increased radius and indicator is helpful

**Comment by maikilangiolo** [comment]
> Woah, woah, woah. That's one too many eggs in one basket. Gorger doesn't even _need_ half of these.

This tbh 💀💀

**Comment by chizzy376** [comment]
Gorger is currently okay, but 3/4 of his kit is useless. This PR makes his kit actually useful, good stuff. Though I'm a bit miffed about the 30u blood drain

**Comment by Araulius** [comment]
marines have unlimited isopills, blood isn't and issue

**Comment by private-tristan** [comment]
Does drain blood drain apply to corpses?

**Comment by Blundir** [comment]
> Does drain blood drain apply to corpses?

No

**Comment by ivanmixo** [comment]
Being about to directly drain marine blood level is problematic due to how severe the hunger slowdown from "low" (like 90%) blood is.

**Comment by Blundir** [comment]
> Being about to directly drain marine blood level is problematic due to how severe the hunger slowdown from "low" (like 90%) blood is.

In practice, to drain a single marine to that point is a challenge on itself, also there is still isotonic everywhere, nano blood for extreme cases, having low blood has more counters than sources at the moment

File: code/datums/status_effects/xeno_buffs.dm [hunk 1]
```diff
@@ -622,8 +623,13 @@
 		X.remove_status_effect(STATUS_EFFECT_XENO_FEAST)
 		return
 	var/heal_amount = X.maxHealth*0.08
-	HEAL_XENO_DAMAGE(X, heal_amount, FALSE)
-	adjustOverheal(X, heal_amount / 2)
+	for(var/mob/living/carbon/xenomorph/affected_xeno AS in GLOB.hive_datums[XENO_HIVE_NORMAL].xenos_by_zlevel["[X.z]"])
```
Comment (Lumipharon) [inline code/datums/status_effects/xeno_buffs.dm:626]: Not that it'll matter often, but this is hard coded to normal hive.

File: code/modules/mob/living/carbon/xenomorph/castes/gorger/abilities_gorger.dm [hunk 2]
```diff
@@ -323,17 +324,13 @@
...
 			to_chat(owner, span_notice("It is beyond our reach, we must be close and our way must be clear."))
 		return FALSE
-	if(HAS_TRAIT(owner, TRAIT_PSY_LINKED))
-		if(!silent)
-			to_chat(owner, span_notice("You are already linked to a xenomorph."))
-		return FALSE
-	if(HAS_TRAIT(target, TRAIT_PSY_LINKED))
-		if(!silent)
-			to_chat(owner, span_notice("[target] is already linked to a xenomorph."))
-		return FALSE
```
Comment (Lumipharon) [inline code/modules/mob/living/carbon/xenomorph/castes/gorger/abilities_gorger.dm:333]: why have these checks been removed?

File: code/modules/mob/living/carbon/xenomorph/castes/gorger/abilities_gorger.dm [hunk 3]
```diff
@@ -416,7 +408,7 @@
 	name = "Feast"
 	action_icon_state = "feast"
 	desc = "Enter a state of rejuvenation. During this time you use a small amount of blood and heal. You can cancel this early."
-	cooldown_timer = 180 SECONDS
+	cooldown_timer = 30 SECONDS
```
Comment (Lumipharon) [inline code/modules/mob/living/carbon/xenomorph/castes/gorger/abilities_gorger.dm:411]: This seems exceptionally low but eh.

**Review by Lumipharon — CHANGES_REQUESTED** [review]

**Comment by Lumipharon** [comment]
Also does feast actually apply to other xenos? I don't see anything in the code, unless I'm missing something.

File: code/datums/status_effects/xeno_buffs.dm [hunk 4]
```diff
@@ -622,8 +623,13 @@
...
+		if(affected_xeno.faction != X.faction)
+			continue
+		if(!line_of_sight(X, affected_xeno, 4) || get_dist(X, affected_xeno) > 4)
+			continue
+		HEAL_XENO_DAMAGE(affected_xeno, heal_amount, FALSE)
+		adjustOverheal(affected_xeno, heal_amount / 2)
```
Comment (Lumipharon) [inline code/datums/status_effects/xeno_buffs.dm:632]: This also seems like, incredibly, mega strong considering there is no AOE/stacking cap, and its going to be very spammable. I am somewhat doubtful of this.

File: code/datums/status_effects/xeno_buffs.dm [hunk 5]
```diff
@@ -622,8 +623,13 @@
 		X.remove_status_effect(STATUS_EFFECT_XENO_FEAST)
 		return
 	var/heal_amount = X.maxHealth*0.08
-	HEAL_XENO_DAMAGE(X, heal_amount, FALSE)
-	adjustOverheal(X, heal_amount / 2)
+	for(var/mob/living/carbon/xenomorph/affected_xeno AS in GLOB.hive_datums[XENO_HIVE_NORMAL].xenos_by_zlevel["[X.z]"])
+		if(affected_xeno.faction != X.faction)
+			continue
+		if(!line_of_sight(X, affected_xeno, 4) || get_dist(X, affected_xeno) > 4)
```
Comment (Lumipharon) [inline code/datums/status_effects/xeno_buffs.dm:629]: is dist even needed if you're already doing LOS with a cap of 4 tiles?

File: code/modules/mob/living/carbon/xenomorph/castes/gorger/abilities_gorger.dm [hunk 6]
```diff
@@ -197,7 +198,7 @@
 
 	if(owner.do_actions)
 		return FALSE
-	if(!line_of_sight(owner, target_xeno, 2) || get_dist(owner, target_xeno) > 2)
+	if(!line_of_sight(owner, target_xeno, 3) || get_dist(owner, target_xeno) > 3)
```
Comment (Lumipharon) [inline code/modules/mob/living/carbon/xenomorph/castes/gorger/abilities_gorger.dm:201]: I know this was existing code, but again, I am fairly sure line_of_sight fails if they're outside of the specified range anyway.

**Review by Lumipharon — CHANGES_REQUESTED** [review]
Feast concerns me, a lot.

**Comment by Aporhtonoma** [comment]
gorger actually draining blood?! Oh hell yea

**Comment by Naaanii** [comment]
Draining real blood lol. Draining 30u lmao.

Gorger powercreep PR of the year.

**Comment by Lumipharon** [comment]
@Blundir There's just too much going on in this PR/very strong changes. Redo it with a single change at a time and it'll be a lot more comfortable to balanc/TM.

**Comment by spookydonut** [comment]
We're happy to consider these changes separated out in to separate PRs but as a package deal this is too big a balance change to consider merging.

**Comment by Blundir** [comment]
Hell nah

**Comment by Arturlang** [comment]
@Blundir this looks great and yeah atomize this more, especially love not having to stop playing the game for psy link

**Pull request closed without merging**